integration/vm_factory/vm_templating_test.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 HyperHQ Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test will perform several tests to validate kata containers with
+# vm templating.
+
+set -e -x
+
+cidir=$(dirname "$0")
+
+source "${cidir}/../../metrics/lib/common.bash"
+
+# Environment variables
+IMAGE="${IMAGE:-busybox}"
+CONTAINER_NAME=$(random_name)
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+
+if [ -f ${DEFAULTCONFIG} ]; then
+	runtime_config_path=${DEFAULTCONFIG}
+else
+	runtime_config_path=${PKGDEFAULTCONFIG}
+fi
+
+kata_runtime_bin=$(command -v kata-runtime)
+
+template_tmpfs_path="/run/vc/vm/template"
+
+enable_vm_template_config() {
+	echo "enable vm template config ${runtime_config_path}"
+	sudo sed -i -e 's/^#\(enable_template\).*=.*$/\1 = true/g' "${runtime_config_path}"
+}
+
+disable_vm_template_config() {
+	echo "disable vm template config"
+	sudo sed -i -e 's/^\(enable_template\).*=.*$/#\1 = true/g' "${runtime_config_path}"
+}
+
+init_vm_template() {
+	echo "init vm template"
+	sudo -E PATH=$PATH "$kata_runtime_bin" factory init
+}
+
+destroy_vm_template() {
+	echo "destroy vm template"
+	sudo -E PATH=$PATH "$kata_runtime_bin" factory destroy
+	# verify template is destroied
+	res=$(mount | grep template | wc -l)
+	test $res -eq 0
```

Comment:
Using `test` is fine (and in a way, makes the code obvious). I'll note though that I don't think this will print an error message in the event of failure will it? Using `[ $res -ne 0 ] || die "a horrid death"` idiom might make it clearer why we died. Not a blocker though, just a thought.